Bluetooth: controller: Fix rejected enc procedure not terminated

joerchan opened the pull request:
Fix issue in the handling of LL_REJECT_EXT_IND packets, this would look
at the procedures that are enqueued, and not the procedure that was
being rejected. This meant that although a reject was received for the
encryption procedure, the handling for a different control procedure was
run.
This would result in the link being terminated as control procedure
timer would time out for the encryption procedure.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

joerchan committed:
Bluetooth: controller: Fix rejected enc procedure not terminated

Fix issue in the handling of LL_REJECT_EXT_IND packets, this would look
at the procedures that are enqueued, and not the procedure that was
being rejected. This meant that although a reject was received for the
encryption procedure, the handling for a different control procedure was
run.
This would result in the link being terminated as control procedure
timer would time out for the encryption procedure.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

joerchan commented on subsys/bluetooth/controller/ll_sw/ctrl.c:
```diff
@@ -2277,55 +2277,60 @@ isr_rx_conn_pkt_ctrl_rej_phy_upd(struct radio_pdu_node_rx *node_rx,
-			_radio.conn_curr->procedure_expire = 0U;
+#if defined(CONFIG_BT_CTLR_PHY)
+	case PDU_DATA_LLCTRL_TYPE_PHY_REQ:
+		if (_radio.conn_curr->llcp_phy.ack !=
```
@cvinayak I think these check should all be llcp_ack != llcp_req && llcp_type != PHY/CONN_PARAM/LENGTH, right?
Could also add one for the ENC procedure?

cvinayak replied:
Please add the check for ENC procedure. For others with state value, I am assuming individual checks are needed (I can rework in a new PR).

joerchan replied:
Added check for the ENC procedure. The other ones I'm not sure about. So I'll leave them as is.

joerchan commented on subsys/bluetooth/controller/ll_sw/ctrl.c:
```diff
@@ -2277,55 +2277,60 @@ isr_rx_conn_pkt_ctrl_rej_phy_upd(struct radio_pdu_node_rx *node_rx,
 static inline void
 isr_rx_conn_pkt_ctrl_rej(struct radio_pdu_node_rx *node_rx, u8_t *rx_enqueue)
```
@cvinayak There is similar code in the SPLIT controller. But I think this bug is a product of the parallel procedure handling. So I'm not certain if the fix should be applied there yet.

joerchan committed:
Bluetooth: controller: Check if enc procedure is in progress

Check if the encryption procedure is in progress when receiving
rejection for the procedure.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

cvinayak commented on subsys/bluetooth/controller/ll_sw/ctrl.c:
```diff
@@ -2274,58 +2274,76 @@ isr_rx_conn_pkt_ctrl_rej_phy_upd(struct radio_pdu_node_rx *node_rx,
 
-		pdu_rx = (void *)node_rx->pdu_data;
-		rej_ext_ind = (void *)&pdu_rx->llctrl.reject_ext_ind;
+	*rx_enqueue = 1U;
```
Move this to after isr_rx_conn_pkt_ctrl_rej_enc call.

cvinayak replied:
the function is reused, hence this ok. resolved.

cvinayak commented on subsys/bluetooth/controller/ll_sw/ctrl.c:
```diff
@@ -2274,58 +2274,76 @@ isr_rx_conn_pkt_ctrl_rej_phy_upd(struct radio_pdu_node_rx *node_rx,
-				rej_ext_ind->error_code;
-			*rx_enqueue = 1U;
-			break;
+			rej_ext_ind->error_code;
```
Move these below isr_rx_conn_pkt_ctrl_rej_enc

cvinayak replied:
missing indentation

cvinayak reviewed (changes requested):
LGTM

cvinayak reviewed (approved):
LGTM

cvinayak reviewed (approved):
LGTM
